qa/tasks/cephfs/test_client_recovery.py

```diff
@@ -808,24 +809,27 @@ def test_client_eviction_if_config_is_set(self):
             # it takes time to have laggy clients entries in cluster log,
             # wait for 6 minutes to see if it is visible, finally restart
             # the client
-            tries = 6
-            while True:
-                try:
-                    with self.assert_cluster_log("1 client(s) laggy due to laggy OSDs",
-                                                 timeout=55):
-                        # make sure clients weren't evicted
-                        self.assert_session_count(2)
-                        break
-                except AssertionError:
-                    tries -= 1
-                    if tries:
-                        continue
-                    raise
+            with contextutil.safe_while(sleep=5, tries=6) as proceed:
+                while proceed():
+                    try:
+                        with self.assert_cluster_log("1 client(s) laggy due to"
+                                                     " laggy OSDs",
+                                                     timeout=55):
+                            # make sure clients weren't evicted
+                            self.assert_session_count(2)
+                            break
+                    except (AssertionError, CommandFailedError) as e:
+                        log.debug(f'{e}, retrying')
+
+            # clear lagginess, expect to get the warning cleared and make sure
+            # client gets evicted
+            self.clear_laggy_params(osd)
+            self.wait_for_health_clear(300)
```

Comment:
Ins't 30s good enough?

Reply:
its not going to wait for 300 always; it runs in a period of 5 seconds; checks if the health cleared else runs continues the loop until elapsed time is 300 seconds; so health got cleared within 30 seconds then it will return

Reply:
I know that. My question was there isn't a need to run for 300s worst case. Max 60s would be more than enough IMO.

Reply:
if health got cleared at X seconds where X is less than 60 and 300 then it doesn't make much difference and in case cluster had multiple warnings and it is recovering from other warnings like Degraded data redundancy or Reduced data availability then might be possible that it takes more than a minute?

Reply:
```
git grep wait_for_health_clear qa/                                                                                                                                         х STOP took 3s ≡ with vshankar@h3ckers-pride
qa/tasks/ceph_test_case.py:    def wait_for_health_clear(self, timeout):
qa/tasks/cephadm_cases/test_cli.py:        self.wait_for_health_clear(60)
qa/tasks/cephadm_cases/test_cli.py:        self.wait_for_health_clear(120)
qa/tasks/cephadm_cases/test_cli.py:        self.wait_for_health_clear(60)
qa/tasks/cephfs/test_auto_repair.py:        self.wait_for_health_clear(timeout=30)
qa/tasks/cephfs/test_failover.py:        self.wait_for_health_clear(30)
qa/tasks/cephfs/test_failover.py:        self.wait_for_health_clear(timeout=30)
qa/tasks/cephfs/test_failover.py:        self.wait_for_health_clear(timeout=30)
qa/tasks/cephfs/test_mantle.py:        self.wait_for_health_clear(timeout=30)
qa/tasks/mgr/dashboard/helper.py:        self.wait_for_health_clear(self.TIMEOUT_HEALTH_CLEAR)
qa/tasks/mgr/dashboard/test_pool.py:        self.wait_for_health_clear(timeout)
qa/tasks/mgr/test_module_selftest.py:        self.wait_for_health_clear(timeout=30)
qa/tasks/mgr/test_orchestrator_cli.py:        self.wait_for_health_clear(30)
```

Should suffice to use 60s.